src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,74 @@
+package seedu.addressbook.commands;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+
+public class EditCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the person identified by the index number used in the last person listing.\n"
+            + "Parameters: INDEX [p]NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS [t/TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2,"
+            + "#02-25 t/friends t/owesMoney";
+
+    public static final String MESSAGE_EDIT_SUCCESS = "Person edited: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+
+    private Person newPerson;
+
+/**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+
+    public EditCommand(int targetVisibleIndex,
+                      String name,
+                      String phone, boolean isPhonePrivate,
+                      String email, boolean isEmailPrivate,
+                      String address, boolean isAddressPrivate,
+                      Set<String> tags) throws IllegalValueException {
+        super(targetVisibleIndex);
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.newPerson = new Person(
+                new Name(name),
+                new Phone(phone, isPhonePrivate),
+                new Email(email, isEmailPrivate),
+                new Address(address, isAddressPrivate),
+                tagSet
+        );
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.removePerson(target);
+            addressBook.addPerson(newPerson);
+            return new CommandResult(String.format(MESSAGE_EDIT_SUCCESS, target));
+
```

Comment:
Stylistic advice: Try to maintain consistency on the use of whitespaces. The blank line here is probably not necessary.